### PR TITLE
KAA-925 Factor out kaa_private.h

### PR DESCRIPTION
**Pull request by rasendubi**

### CODE REVIEW

File: client/client-multi/client-c/test/collections/test_kaa_list.c [hunk 1]
```diff
@@ -105,7 +105,7 @@ void test_list_push_back()
         kaa_list_push_back(list, number);
     }
 
-    ASSERT_EQUAL(kaa_list_get_size(list), node_number);
+    ASSERT_EQUAL(kaa_list_get_size(list), (size_t)node_number);
```
Comment (forGGe) [inline client/client-multi/client-c/test/collections/test_kaa_list.c]: Here:

**Comment by ghost** [comment]
LGTM

**Comment by forGGe** [comment]
Looks good

**Pull request merged**